### ISSUE TITLE
Overfitting problem when training transformer

### ISSUE THREAD

**Issue by fnzhan** [CompVis/taming-transformers#48]
I train the transformer but find it overfits after 30-40 epochs, with the validation loss goes high and the training loss is very small.
If you meet this problem in model training. Now I try to use the pkeep=0.9 in the cond_transformer.py to avoid overfitting.

**Comment by ink1** [comment]
I initially thought that the 256x256 training images are randomly cropped from the dataset you provide. But that turned out to be not the case and I had to provide a set of 256x256 crops myself. I see no reason why random cropping should not work (other than slightly higher overhead) and it may help with overfitting but I did not get to the bottom of this yet.

**Comment by fnzhan** [comment]
For my part, overfitting is caused by the removal of random crop. After adding the random crop in model training, the validation loss goes well.

**Comment by adeptflax** [comment]
Could adding random crop to the training data improve my image2image model. https://github.com/CompVis/taming-transformers/issues/51 I based the code off of imagenet

**Comment by adeptflax** [comment]
I have around 11k training examples.

**Comment by adeptflax** [comment]
Would there a way to do pre-training?

**Comment by LeeDoYup** [comment]
I also think that the stage 2 training of VQ-GAN would suffer from the overfitting on FFHQ, because this repository does not contain data augmentations for FFHQ training dataset.

**Comment by MHVali** [comment]
@fnzhan 
Thanks for your comment in this issue! I want to train the transformer using a trained VQ-VAE on FFHQ dataset. I used the code from [this repo](https://github.com/dome272/VQGAN-pytorch.git). In the repo, they use pkeep=0.5 when training the transformer. But, I don't have any idea what could be a good pkeep value for training the transformer on FFHQ dataset, in order not to overfit.

I was also thinking of using a scheduled masking by having pkeep=0.1 in the beginning of the training and increasing it to pkeep=0.9 at the end of training. What is your recommendation for setting pkeep?

It is not important for me to get FID values as good as the VQGAN paper. I only want to have reasonable and intelligible generations. Thanks in advance for your help!
